Cabal 2.0.0.2 incompatibility

The build failure is 
```
==> cabal install --jobs=8 --max-backjumps=100000 --prefix=/usr/local/Cellar/idris/1.0.1 -f FFI -f release
clang: warning: -Wl,-headerpad_max_install_names: 'linker' input unused
clang: warning: argument unused during compilation: '-L/usr/local/opt/libffi/lib'
clang: warning: argument unused during compilation: '-L/usr/local/lib'
clang: warning: argument unused during compilation: '-L/System/Library/Frameworks/OpenGL.framework/Versions/Current/Libraries'
Resolving dependencies...
Notice: installing into a sandbox located at
/private/tmp/idris-20170801-1139-1ykhe4s/Idris-dev-1.0.1/.cabal-sandbox
Configuring idris-1.0.1...
Failed to install idris-1.0.1
Build log ( /private/tmp/idris-20170801-1139-1ykhe4s/Idris-dev-1.0.1/.cabal-sandbox/logs/idris-1.0.1.log ):
cabal: Entering directory '.'
cabal: Leaving directory '.'
cabal: Error: some packages failed to install:
idris-1.0.1 failed during the configure step. The exception was:
user error ('/usr/local/opt/ghc/bin/ghc' exited with an error:

dist/dist-sandbox-15a7f92e/setup/setup.hs:59:16: error:
• Data constructor not in scope: FlagName :: [Char] -> FlagName
• Perhaps you meant one of these:
variable ‘flagName’ (imported from Distribution.PackageDescription),
variable ‘mkFlagName’ (imported from Distribution.PackageDescription),
variable ‘unFlagName’ (imported from Distribution.PackageDescription)
|
59 | case lookup (FlagName "gmp") (S.configConfigurationsFlags flags) of
| ^^^^^^^^

dist/dist-sandbox-15a7f92e/setup/setup.hs:66:16: error:
• Data constructor not in scope: FlagName :: [Char] -> FlagName
• Perhaps you meant one of these:
variable ‘flagName’ (imported from Distribution.PackageDescription),
variable ‘mkFlagName’ (imported from Distribution.PackageDescription),
variable ‘unFlagName’ (imported from Distribution.PackageDescription)
|
66 | case lookup (FlagName "execonly") (S.configConfigurationsFlags flags) of
| ^^^^^^^^

dist/dist-sandbox-15a7f92e/setup/setup.hs:73:18: error:
• Data constructor not in scope: FlagName :: [Char] -> FlagName
• Perhaps you meant one of these:
variable ‘flagName’ (imported from Distribution.PackageDescription),
variable ‘mkFlagName’ (imported from Distribution.PackageDescription),
variable ‘unFlagName’ (imported from Distribution.PackageDescription)
|
73 | case lookup (FlagName "release") (S.configConfigurationsFlags flags) of
| ^^^^^^^^

dist/dist-sandbox-15a7f92e/setup/setup.hs:80:16: error:
• Data constructor not in scope: FlagName :: [Char] -> FlagName
• Perhaps you meant one of these:
variable ‘flagName’ (imported from Distribution.PackageDescription),
variable ‘mkFlagName’ (imported from Distribution.PackageDescription),
variable ‘unFlagName’ (imported from Distribution.PackageDescription)
|
80 | case lookup (FlagName "freestanding") (S.configConfigurationsFlags flags)
of
| ^^^^^^^^
)
```

Note the deprecations

> - Backwards incompatible change to 'FlagName' (#4062)
> - Backwards incompatible change to 'Version' (#3905)

here:
https://github.com/haskell/cabal/blob/2d3135722fa08294d3425b3f9333e124c1f48de1/Cabal/changelog#L59

(To get this far, you first need to add allow-newer for base,process,time,trifecta if you're using GHC 8.2.1)

@ahmadsalim nope:
```
==> Cloning https://github.com/idris-lang/Idris-dev.git
Updating /Users/joe/Library/Caches/Homebrew/idris--git
git config remote.origin.url https://github.com/idris-lang/Idris-dev.git
git config remote.origin.fetch +refs/heads/master:refs/remotes/origin/master
git fetch origin
==> Checking out branch master
git checkout -f master --
Already on 'master'
Your branch is up-to-date with 'origin/master'.
git reset --hard origin/master
HEAD is now at 3d5f454 Merge pull request #3929 from clayrat/add-tactics
==> cabal sandbox init
Config file path source is default config file.
Config file /private/tmp/idris-20170801-79490-3yqvm/.cabal/config not found.
Writing default configuration to
/private/tmp/idris-20170801-79490-3yqvm/.cabal/config
Writing a default package environment file to
/private/tmp/idris-20170801-79490-3yqvm/cabal.sandbox.config
Creating a new sandbox at
/private/tmp/idris-20170801-79490-3yqvm/.cabal-sandbox
==> cabal update
Downloading the latest package list from hackage.haskell.org
==> cabal install --jobs=8 --max-backjumps=100000 Cabal
clang: warning: -Wl,-headerpad_max_install_names: 'linker' input unused
clang: warning: argument unused during compilation: '-L/usr/local/opt/libffi/lib'
clang: warning: argument unused during compilation: '-L/usr/local/lib'
clang: warning: argument unused during compilation: '-L/System/Library/Frameworks/OpenGL.framework/Versions/Current/Libraries'
Resolving dependencies...
Notice: installing into a sandbox located at
/private/tmp/idris-20170801-79490-3yqvm/.cabal-sandbox
Downloading process-1.4.3.0...
Configuring process-1.4.3.0...
Building process-1.4.3.0...
Installed process-1.4.3.0
Downloading Cabal-1.24.2.0...
Configuring Cabal-1.24.2.0...
Building Cabal-1.24.2.0...
Installed Cabal-1.24.2.0
==> cabal install --jobs=8 --max-backjumps=100000 --only-dependencies -f FFI
clang: warning: -Wl,-headerpad_max_install_names: 'linker' input unused
clang: warning: argument unused during compilation: '-L/usr/local/opt/libffi/lib'
clang: warning: argument unused during compilation: '-L/usr/local/lib'
clang: warning: argument unused during compilation: '-L/System/Library/Frameworks/OpenGL.framework/Versions/Current/Libraries'
Resolving dependencies...
Notice: installing into a sandbox located at
/private/tmp/idris-20170801-79490-3yqvm/.cabal-sandbox
Downloading annotated-wl-pprint-0.7.0...
Downloading ansi-terminal-0.6.3.1...
Configuring process-1.4.3.0...
Configuring annotated-wl-pprint-0.7.0...
Downloading base-compat-0.9.3...
Configuring ansi-terminal-0.6.3.1...
Downloading base-orphans-0.6...
Configuring base-compat-0.9.3...
Downloading base64-bytestring-1.0.0.1...
Downloading cereal-0.5.4.0...
Configuring base64-bytestring-1.0.0.1...
Configuring base-orphans-0.6...
Downloading code-page-0.1.3...
Configuring code-page-0.1.3...
Downloading data-default-class-0.1.2.0...
Configuring cereal-0.5.4.0...
Downloading digest-0.0.1.2...
Building annotated-wl-pprint-0.7.0...
Downloading dlist-0.8.0.3...
Downloading fail-4.9.0.0...
Building ansi-terminal-0.6.3.1...
Downloading fingertree-0.1.1.0...
Building base-compat-0.9.3...
Downloading ieee754-0.8.0...
Building base64-bytestring-1.0.0.1...
Downloading integer-logarithms-1.0.2...
Building base-orphans-0.6...
Downloading libffi-0.1...
Downloading mtl-2.2.1...
Building code-page-0.1.3...
Downloading network-2.6.3.2...
Building cereal-0.5.4.0...
Downloading old-locale-1.0.0.7...
Downloading parallel-3.2.1.1...
Downloading prelude-extras-0.4.0.3...
Downloading primitive-0.6.2.0...
Downloading random-1.1...
Downloading reflection-2.1.2...
Downloading safe-0.3.15...
Downloading semigroups-0.18.3...
Downloading split-0.2.3.2...
Installed base-orphans-0.6
Configuring digest-0.0.1.2...
Downloading stm-2.4.4.1...
Downloading syb-0.7...
Downloading terminal-size-0.3.2.1...
Downloading text-1.2.2.2...
Configuring data-default-class-0.1.2.0...
Installed annotated-wl-pprint-0.7.0
Downloading th-abstraction-0.2.4.0...
Downloading time-locale-compat-0.1.1.3...
Configuring dlist-0.8.0.3...
Installed code-page-0.1.3
Downloading transformers-compat-0.5.1.4...
Downloading unix-compat-0.4.3.1...
Downloading utf8-string-1.0.1.1...
Downloading void-0.7.2...
Building digest-0.0.1.2...
Downloading zlib-0.6.1.2...
Configuring fail-4.9.0.0...
Installed base64-bytestring-1.0.0.1
Building data-default-class-0.1.2.0...
Building dlist-0.8.0.3...
Building fail-4.9.0.0...
Configuring fingertree-0.1.1.0...
Installed data-default-class-0.1.2.0
Downloading data-default-instances-containers-0.0.1...
Building process-1.4.3.0...
Configuring ieee754-0.8.0...
Installed fail-4.9.0.0
Configuring integer-logarithms-1.0.2...
Installed digest-0.0.1.2
Configuring libffi-0.1...
Installed dlist-0.8.0.3
Downloading data-default-instances-dlist-0.0.1...
Building fingertree-0.1.1.0...
Configuring mtl-2.2.1...
Installed ansi-terminal-0.6.3.1
Downloading ansi-wl-pprint-0.6.7.3...
Building ieee754-0.8.0...
Building integer-logarithms-1.0.2...
Building libffi-0.1...
Building mtl-2.2.1...
Configuring old-locale-1.0.0.7...
Installed base-compat-0.9.3
Building old-locale-1.0.0.7...
Configuring network-2.6.3.2...
Installed integer-logarithms-1.0.2
Configuring parallel-3.2.1.1...
Installed process-1.4.3.0
Configuring primitive-0.6.2.0...
Installed libffi-0.1
Building parallel-3.2.1.1...
Configuring prelude-extras-0.4.0.3...
Installed old-locale-1.0.0.7
Downloading data-default-instances-old-locale-0.0.1...
Downloading old-time-1.1.0.3...
Building primitive-0.6.2.0...
Configuring random-1.1...
Installed ieee754-0.8.0
Building prelude-extras-0.4.0.3...
Building random-1.1...
Configuring safe-0.3.15...
Installed fingertree-0.1.1.0
Configuring reflection-2.1.2...
Installed mtl-2.2.1
Downloading regex-base-0.93.2...
Building safe-0.3.15...
Installed parallel-3.2.1.1
Configuring semigroups-0.18.3...
Building reflection-2.1.2...
Building semigroups-0.18.3...
Configuring split-0.2.3.2...
Installed cereal-0.5.4.0
Configuring stm-2.4.4.1...
Installed prelude-extras-0.4.0.3
Configuring terminal-size-0.3.2.1...
Installed semigroups-0.18.3
Building split-0.2.3.2...
Building stm-2.4.4.1...
Building terminal-size-0.3.2.1...
Configuring syb-0.7...
Installed safe-0.3.15
Configuring text-1.2.2.2...
Installed split-0.2.3.2
Building syb-0.7...
Configuring th-abstraction-0.2.4.0...
Installed random-1.1
Building text-1.2.2.2...
Configuring time-locale-compat-0.1.1.3...
Building th-abstraction-0.2.4.0...
Installed terminal-size-0.3.2.1
Building network-2.6.3.2...
Configuring transformers-compat-0.5.1.4...
Installed stm-2.4.4.1
Downloading StateVar-1.1.0.4...
Downloading async-2.1.1.1...
Building time-locale-compat-0.1.1.3...
Configuring unix-compat-0.4.3.1...
Installed reflection-2.1.2
Building transformers-compat-0.5.1.4...
Configuring void-0.7.2...
Installed primitive-0.6.2.0
Downloading vector-0.12.0.1...
Configuring utf8-string-1.0.1.1...
Installed time-locale-compat-0.1.1.3
Building unix-compat-0.4.3.1...
Building void-0.7.2...
Building utf8-string-1.0.1.1...
Configuring zlib-0.6.1.2...
Installed transformers-compat-0.5.1.4
Downloading exceptions-0.8.3...
Downloading tagged-0.8.5...
Configuring data-default-instances-containers-0.0.1...
Installed syb-0.7
Configuring data-default-instances-dlist-0.0.1...
Installed void-0.7.2
Building zlib-0.6.1.2...
Building data-default-instances-containers-0.0.1...
Building data-default-instances-dlist-0.0.1...
Configuring ansi-wl-pprint-0.6.7.3...
Installed unix-compat-0.4.3.1
Configuring Cabal-1.24.2.0...
Installed data-default-instances-containers-0.0.1
Configuring old-time-1.1.0.3...
Installed data-default-instances-dlist-0.0.1
Building ansi-wl-pprint-0.6.7.3...
Configuring data-default-instances-old-locale-0.0.1...
Installed th-abstraction-0.2.4.0
Building Cabal-1.24.2.0...
Building data-default-instances-old-locale-0.0.1...
Configuring regex-base-0.93.2...
Installed utf8-string-1.0.1.1
Installed data-default-instances-old-locale-0.0.1
Downloading data-default-0.7.1.1...
Configuring async-2.1.1.1...
Configuring StateVar-1.1.0.4...
Installed ansi-wl-pprint-0.6.7.3
Downloading optparse-applicative-0.14.0.0...
Building regex-base-0.93.2...
Building async-2.1.1.1...
Building StateVar-1.1.0.4...
Configuring vector-0.12.0.1...
Installed zlib-0.6.1.2
Configuring exceptions-0.8.3...
Installed StateVar-1.1.0.4
Downloading contravariant-1.4...
Configuring tagged-0.8.5...
Installed regex-base-0.93.2
Building vector-0.12.0.1...
Building old-time-1.1.0.3...
Building exceptions-0.8.3...
Installed async-2.1.1.1
Configuring data-default-0.7.1.1...
Building tagged-0.8.5...
Building data-default-0.7.1.1...
Configuring optparse-applicative-0.14.0.0...
Installed network-2.6.3.2
Configuring contravariant-1.4...
Installed data-default-0.7.1.1
Building optparse-applicative-0.14.0.0...
Installed exceptions-0.8.3
Building contravariant-1.4...
Installed tagged-0.8.5
Installed old-time-1.1.0.3
Installed contravariant-1.4
Installed optparse-applicative-0.14.0.0
Installed text-1.2.2.2
Downloading blaze-builder-0.4.0.2...
Downloading hashable-1.2.6.1...
Downloading hfsevents-0.1.6...
Configuring hashable-1.2.6.1...
Downloading parsec-3.1.11...
Configuring blaze-builder-0.4.0.2...
Configuring parsec-3.1.11...
Downloading tagsoup-0.14.1...
Configuring hfsevents-0.1.6...
Downloading zip-archive-0.3.1.1...
Configuring tagsoup-0.14.1...
Configuring zip-archive-0.3.1.1...
Building hashable-1.2.6.1...
Building blaze-builder-0.4.0.2...
Building parsec-3.1.11...
Building hfsevents-0.1.6...
Building tagsoup-0.14.1...
Building zip-archive-0.3.1.1...
Installed hfsevents-0.1.6
Downloading fsnotify-0.2.1.1...
Configuring fsnotify-0.2.1.1...
Building fsnotify-0.2.1.1...
Installed hashable-1.2.6.1
Downloading scientific-0.3.5.1...
Downloading unordered-containers-0.2.8.0...
Configuring unordered-containers-0.2.8.0...
Downloading uuid-types-1.0.3...
Installed blaze-builder-0.4.0.2
Downloading blaze-markup-0.8.0.0...
Configuring scientific-0.3.5.1...
Building unordered-containers-0.2.8.0...
Building scientific-0.3.5.1...
Configuring uuid-types-1.0.3...
Installed zip-archive-0.3.1.1
Building uuid-types-1.0.3...
Installed fsnotify-0.2.1.1
Configuring blaze-markup-0.8.0.0...
Building blaze-markup-0.8.0.0...
Installed uuid-types-1.0.3
Installed scientific-0.3.5.1
Downloading attoparsec-0.13.2.0...
Installed parsec-3.1.11
Downloading network-uri-2.6.1.0...
Downloading regex-tdfa-1.2.2...
Configuring attoparsec-0.13.2.0...
Configuring network-uri-2.6.1.0...
Configuring regex-tdfa-1.2.2...
Building attoparsec-0.13.2.0...
Building network-uri-2.6.1.0...
Building regex-tdfa-1.2.2...
Installed blaze-markup-0.8.0.0
Downloading blaze-html-0.9.0.1...
Configuring blaze-html-0.9.0.1...
Installed tagsoup-0.14.1
Building blaze-html-0.9.0.1...
Installed unordered-containers-0.2.8.0
Downloading charset-0.3.7.1...
Downloading uniplate-1.6.12...
Configuring uniplate-1.6.12...
Configuring charset-0.3.7.1...
Building uniplate-1.6.12...
Installed network-uri-2.6.1.0
Building charset-0.3.7.1...
Installed vector-0.12.0.1
Downloading vector-binary-instances-0.2.3.5...
Configuring vector-binary-instances-0.2.3.5...
Building vector-binary-instances-0.2.3.5...
Installed vector-binary-instances-0.2.3.5
Installed uniplate-1.6.12
Installed blaze-html-0.9.0.1
Installed charset-0.3.7.1
Installed attoparsec-0.13.2.0
Downloading aeson-1.2.1.0...
Downloading css-text-0.1.2.2...
Configuring css-text-0.1.2.2...
Configuring aeson-1.2.1.0...
Building css-text-0.1.2.2...
Building aeson-1.2.1.0...
Installed css-text-0.1.2.2
Downloading xss-sanitize-0.3.5.7...
Configuring xss-sanitize-0.3.5.7...
Building xss-sanitize-0.3.5.7...
Installed xss-sanitize-0.3.5.7
Downloading cheapskate-0.1.1...
Configuring cheapskate-0.1.1...
Building cheapskate-0.1.1...
Installed cheapskate-0.1.1
Installed Cabal-1.24.2.0
Downloading cabal-doctest-1.0.2...
Configuring cabal-doctest-1.0.2...
Building cabal-doctest-1.0.2...
Installed cabal-doctest-1.0.2
Downloading distributive-0.5.3...
Downloading parsers-0.12.6...
Configuring distributive-0.5.3...
Configuring parsers-0.12.6...
Building distributive-0.5.3...
Installed regex-tdfa-1.2.2
Building parsers-0.12.6...
Installed distributive-0.5.3
Downloading comonad-5.0.2...
Configuring comonad-5.0.2...
Building comonad-5.0.2...
Installed aeson-1.2.1.0
Installed parsers-0.12.6
Installed comonad-5.0.2
Downloading bifunctors-5.4.2...
Configuring bifunctors-5.4.2...
Building bifunctors-5.4.2...
Installed bifunctors-5.4.2
Downloading profunctors-5.2.1...
Downloading semigroupoids-5.2.1...
Configuring semigroupoids-5.2.1...
Configuring profunctors-5.2.1...
Building profunctors-5.2.1...
Building semigroupoids-5.2.1...
Installed profunctors-5.2.1
Installed semigroupoids-5.2.1
Downloading free-4.12.4...
Downloading reducers-3.12.1...
Configuring reducers-3.12.1...
Configuring free-4.12.4...
Building reducers-3.12.1...
Building free-4.12.4...
Installed reducers-3.12.1
Installed free-4.12.4
Downloading adjunctions-4.3...
Configuring adjunctions-4.3...
Building adjunctions-4.3...
Installed adjunctions-4.3
Downloading kan-extensions-5.0.2...
Configuring kan-extensions-5.0.2...
Building kan-extensions-5.0.2...
Installed kan-extensions-5.0.2
Downloading lens-4.15.3...
Configuring lens-4.15.3...
Building lens-4.15.3...
Installed lens-4.15.3
Downloading trifecta-1.7.1...
Configuring trifecta-1.7.1...
Building trifecta-1.7.1...
Installed trifecta-1.7.1
==> cabal install --jobs=8 --max-backjumps=100000 --prefix=/usr/local/Cellar/idris/HEAD-3d5f454 -f FFI
clang: warning: -Wl,-headerpad_max_install_names: 'linker' input unused
clang: warning: argument unused during compilation: '-L/usr/local/opt/libffi/lib'
clang: warning: argument unused during compilation: '-L/usr/local/lib'
clang: warning: argument unused during compilation: '-L/System/Library/Frameworks/OpenGL.framework/Versions/Current/Libraries'
Resolving dependencies...
In order, the following will be installed:
haskeline-0.7.4.0 (via: idris-1.1.0 idris-1.1.0 idris-1.1.0 idris-1.1.0 idris-1.1.0) (reinstall) (changes: process-1.6.1.0 -> 1.4.3.0)
optparse-applicative-0.14.0.0 (via: idris-1.1.0) (reinstall) (changes: process-1.6.1.0 -> 1.4.3.0)
idris-1.1.0 +ffi (new package)
Warning: Note that reinstalls are always dangerous. Continuing anyway...
Notice: installing into a sandbox located at
/private/tmp/idris-20170801-79490-3yqvm/.cabal-sandbox
Downloading haskeline-0.7.4.0...
Configuring optparse-applicative-0.14.0.0...
Configuring haskeline-0.7.4.0...
Building optparse-applicative-0.14.0.0...
Building haskeline-0.7.4.0...
Installed optparse-applicative-0.14.0.0
Installed haskeline-0.7.4.0
Configuring idris-1.1.0...
Building idris-1.1.0...
Failed to install idris-1.1.0
Build log ( /private/tmp/idris-20170801-79490-3yqvm/.cabal-sandbox/logs/idris-1.1.0.log ):
cabal: Entering directory '.'
[1 of 1] Compiling Main             ( dist/dist-sandbox-de2ad256/setup/setup.hs, dist/dist-sandbox-de2ad256/setup/Main.o )
Linking ./dist/dist-sandbox-de2ad256/setup/setup ...
Configuring idris-1.1.0...
Warning: Packages using 'cabal-version: >= 1.10' must specify the
'default-language' field for each component (e.g. Haskell98 or Haskell2010).
If a component uses different languages in different modules then list the
other ones in the 'other-languages' field.
clang: warning: -Wl,-headerpad_max_install_names: 'linker' input unused
clang: warning: argument unused during compilation: '-L/usr/local/opt/libffi/lib'
clang: warning: argument unused during compilation: '-L/usr/local/lib'
clang: warning: argument unused during compilation: '-L/System/Library/Frameworks/OpenGL.framework/Versions/Current/Libraries'
rm -f idris_rts.o idris_heap.o idris_gc.o idris_gmp.o idris_bitstring.o idris_opts.o idris_stats.o idris_utf8.o idris_stdfgn.o mini-gmp.o idris_buffer.o getline.o idris_net.o libidris_rts.a 
Generating dist/dist-sandbox-de2ad256/build/autogen/Version_idris.hs for prerelease 3d5f454
Building idris-1.1.0...
Preprocessing library idris-1.1.0...
[  1 of 114] Compiling IRTS.JavaScript.AST ( src/IRTS/JavaScript/AST.hs, dist/dist-sandbox-de2ad256/build/IRTS/JavaScript/AST.o )
[  2 of 114] Compiling Idris.Colours    ( src/Idris/Colours.hs, dist/dist-sandbox-de2ad256/build/Idris/Colours.o )
[  3 of 114] Compiling Idris.Help       ( src/Idris/Help.hs, dist/dist-sandbox-de2ad256/build/Idris/Help.o )
[  4 of 114] Compiling Paths_idris      ( dist/dist-sandbox-de2ad256/build/autogen/Paths_idris.hs, dist/dist-sandbox-de2ad256/build/Paths_idris.o )
[  5 of 114] Compiling IRTS.System      ( src/IRTS/System.hs, dist/dist-sandbox-de2ad256/build/IRTS/System.o )
[  6 of 114] Compiling Tools_idris      ( dist/dist-sandbox-de2ad256/build/autogen/Tools_idris.hs, dist/dist-sandbox-de2ad256/build/Tools_idris.o )
[  7 of 114] Compiling Util.DynamicLinker ( src/Util/DynamicLinker.hs, dist/dist-sandbox-de2ad256/build/Util/DynamicLinker.o )
[  8 of 114] Compiling Util.Net         ( src/Util/Net.hs, dist/dist-sandbox-de2ad256/build/Util/Net.o )
[  9 of 114] Compiling Util.Pretty      ( src/Util/Pretty.hs, dist/dist-sandbox-de2ad256/build/Util/Pretty.o )
[ 10 of 114] Compiling Idris.Core.TT    ( src/Idris/Core/TT.hs, dist/dist-sandbox-de2ad256/build/Idris/Core/TT.o )
[ 11 of 114] Compiling Idris.Unlit      ( src/Idris/Unlit.hs, dist/dist-sandbox-de2ad256/build/Idris/Unlit.o )
[ 12 of 114] Compiling Idris.Docstrings ( src/Idris/Docstrings.hs, dist/dist-sandbox-de2ad256/build/Idris/Docstrings.o )
[ 13 of 114] Compiling Idris.Core.Constraints ( src/Idris/Core/Constraints.hs, dist/dist-sandbox-de2ad256/build/Idris/Core/Constraints.o )
[ 14 of 114] Compiling Idris.Core.CaseTree ( src/Idris/Core/CaseTree.hs, dist/dist-sandbox-de2ad256/build/Idris/Core/CaseTree.o )
[ 15 of 114] Compiling Idris.Core.Evaluate ( src/Idris/Core/Evaluate.hs, dist/dist-sandbox-de2ad256/build/Idris/Core/Evaluate.o )
[ 16 of 114] Compiling Idris.Core.WHNF  ( src/Idris/Core/WHNF.hs, dist/dist-sandbox-de2ad256/build/Idris/Core/WHNF.o )
[ 17 of 114] Compiling Idris.Core.Unify ( src/Idris/Core/Unify.hs, dist/dist-sandbox-de2ad256/build/Idris/Core/Unify.o )
[ 18 of 114] Compiling Idris.Core.Typecheck ( src/Idris/Core/Typecheck.hs, dist/dist-sandbox-de2ad256/build/Idris/Core/Typecheck.o )
[ 19 of 114] Compiling Idris.Core.ProofTerm ( src/Idris/Core/ProofTerm.hs, dist/dist-sandbox-de2ad256/build/Idris/Core/ProofTerm.o )
[ 20 of 114] Compiling Idris.Core.DeepSeq ( src/Idris/Core/DeepSeq.hs, dist/dist-sandbox-de2ad256/build/Idris/Core/DeepSeq.o )
[ 21 of 114] Compiling Idris.Core.Binary ( src/Idris/Core/Binary.hs, dist/dist-sandbox-de2ad256/build/Idris/Core/Binary.o )
[ 22 of 114] Compiling Idris.IdeMode    ( src/Idris/IdeMode.hs, dist/dist-sandbox-de2ad256/build/Idris/IdeMode.o )
[ 23 of 114] Compiling IRTS.Lang        ( src/IRTS/Lang.hs, dist/dist-sandbox-de2ad256/build/IRTS/Lang.o )
[ 24 of 114] Compiling IRTS.LangOpts    ( src/IRTS/LangOpts.hs, dist/dist-sandbox-de2ad256/build/IRTS/LangOpts.o )
[ 25 of 114] Compiling IRTS.JavaScript.Specialize ( src/IRTS/JavaScript/Specialize.hs, dist/dist-sandbox-de2ad256/build/IRTS/JavaScript/Specialize.o )
[ 26 of 114] Compiling IRTS.JavaScript.PrimOp ( src/IRTS/JavaScript/PrimOp.hs, dist/dist-sandbox-de2ad256/build/IRTS/JavaScript/PrimOp.o )
[ 27 of 114] Compiling IRTS.JavaScript.Name ( src/IRTS/JavaScript/Name.hs, dist/dist-sandbox-de2ad256/build/IRTS/JavaScript/Name.o )
[ 28 of 114] Compiling IRTS.JavaScript.LangTransforms ( src/IRTS/JavaScript/LangTransforms.hs, dist/dist-sandbox-de2ad256/build/IRTS/JavaScript/LangTransforms.o )
[ 29 of 114] Compiling IRTS.Defunctionalise ( src/IRTS/Defunctionalise.hs, dist/dist-sandbox-de2ad256/build/IRTS/Defunctionalise.o )
[ 30 of 114] Compiling IRTS.Simplified  ( src/IRTS/Simplified.hs, dist/dist-sandbox-de2ad256/build/IRTS/Simplified.o )
[ 31 of 114] Compiling IRTS.Inliner     ( src/IRTS/Inliner.hs, dist/dist-sandbox-de2ad256/build/IRTS/Inliner.o )
[ 32 of 114] Compiling IRTS.CodegenCommon ( src/IRTS/CodegenCommon.hs, dist/dist-sandbox-de2ad256/build/IRTS/CodegenCommon.o )
[ 33 of 114] Compiling Idris.Options    ( src/Idris/Options.hs, dist/dist-sandbox-de2ad256/build/Idris/Options.o )
[ 34 of 114] Compiling Idris.Package.Common ( src/Idris/Package/Common.hs, dist/dist-sandbox-de2ad256/build/Idris/Package/Common.o )
[ 35 of 114] Compiling IRTS.JavaScript.Codegen ( src/IRTS/JavaScript/Codegen.hs, dist/dist-sandbox-de2ad256/build/IRTS/JavaScript/Codegen.o )
[ 36 of 114] Compiling IRTS.CodegenJavaScript ( src/IRTS/CodegenJavaScript.hs, dist/dist-sandbox-de2ad256/build/IRTS/CodegenJavaScript.o )
[ 37 of 114] Compiling IRTS.Bytecode    ( src/IRTS/Bytecode.hs, dist/dist-sandbox-de2ad256/build/IRTS/Bytecode.o )
[ 38 of 114] Compiling IRTS.Portable    ( src/IRTS/Portable.hs, dist/dist-sandbox-de2ad256/build/IRTS/Portable.o )
[ 39 of 114] Compiling IRTS.DumpBC      ( src/IRTS/DumpBC.hs, dist/dist-sandbox-de2ad256/build/IRTS/DumpBC.o )
[ 40 of 114] Compiling IRTS.BCImp       ( src/IRTS/BCImp.hs, dist/dist-sandbox-de2ad256/build/IRTS/BCImp.o )
[ 41 of 114] Compiling Idris.Core.ProofState ( src/Idris/Core/ProofState.hs, dist/dist-sandbox-de2ad256/build/Idris/Core/ProofState.o )
[ 42 of 114] Compiling Idris.Core.Elaborate ( src/Idris/Core/Elaborate.hs, dist/dist-sandbox-de2ad256/build/Idris/Core/Elaborate.o )
[ 43 of 114] Compiling Idris.AbsSyntaxTree ( src/Idris/AbsSyntaxTree.hs, dist/dist-sandbox-de2ad256/build/Idris/AbsSyntaxTree.o )
[ 44 of 114] Compiling Idris.REPL.Commands ( src/Idris/REPL/Commands.hs, dist/dist-sandbox-de2ad256/build/Idris/REPL/Commands.o )
[ 45 of 114] Compiling Idris.DeepSeq    ( src/Idris/DeepSeq.hs, dist/dist-sandbox-de2ad256/build/Idris/DeepSeq.o )

src/Idris/DeepSeq.hs:32:10: error:
    Duplicate instance declarations:
      instance NFData CT.Options -- Defined at src/Idris/DeepSeq.hs:32:10
      instance NFData CT.Options -- Defined in ‘Cheapskate.Types’
   |
32 | instance NFData CT.Options where
   |          ^^^^^^^^^^^^^^^^^

src/Idris/DeepSeq.hs:35:10: error:
    Duplicate instance declarations:
      instance NFData CT.ListType
        -- Defined at src/Idris/DeepSeq.hs:35:10
      instance NFData CT.ListType -- Defined in ‘Cheapskate.Types’
   |
35 | instance NFData CT.ListType where
   |          ^^^^^^^^^^^^^^^^^^

src/Idris/DeepSeq.hs:39:10: error:
    Duplicate instance declarations:
      instance NFData CT.CodeAttr
        -- Defined at src/Idris/DeepSeq.hs:39:10
      instance NFData CT.CodeAttr -- Defined in ‘Cheapskate.Types’
   |
39 | instance NFData CT.CodeAttr where
   |          ^^^^^^^^^^^^^^^^^^

src/Idris/DeepSeq.hs:42:10: error:
    Duplicate instance declarations:
      instance NFData CT.NumWrapper
        -- Defined at src/Idris/DeepSeq.hs:42:10
      instance NFData CT.NumWrapper -- Defined in ‘Cheapskate.Types’
   |
42 | instance NFData CT.NumWrapper where
   |          ^^^^^^^^^^^^^^^^^^^^
cabal: Leaving directory '.'
cabal: Error: some packages failed to install:
idris-1.1.0 failed during the building phase. The exception was:
ExitFailure 1
/usr/local/Homebrew/Library/Homebrew/build.rb (Formulary::FormulaLoader): loading /usr/local/Homebrew/Library/Taps/homebrew/homebrew-core/Formula/python.rb
/usr/local/Homebrew/Library/Homebrew/debrew.rb:11:in `raise'
BuildError: Failed executing: cabal install --jobs=8 --max-backjumps=100000 --prefix=/usr/local/Cellar/idris/HEAD-3d5f454 -f\ FFI
1. raise
2. ignore
3. backtrace
4. irb
5. shell
Choose an action:
```

--- a/Setup.hs
+++ b/Setup.hs
@@ -5,7 +5,7 @@ import Data.IORef
 import Control.Exception (SomeException, catch)
 
 import Distribution.Simple
-import Distribution.Simple.BuildPaths (autogenModulesDir)
+import Distribution.Simple.BuildPaths
 import Distribution.Simple.InstallDirs as I
 import Distribution.Simple.LocalBuildInfo as L
 import qualified Distribution.Simple.Setup as S
@@ -56,31 +56,36 @@ windres verbosity = P.runProgramInvocation verbosity . P.simpleProgramInvocation
 
 usesGMP :: S.ConfigFlags -> Bool
 usesGMP flags =
-  case lookup (FlagName "gmp") (S.configConfigurationsFlags flags) of
+  case lookup (mkFlagName "gmp") (S.configConfigurationsFlags flags) of
     Just True -> True
     Just False -> False
     Nothing -> False
 
 execOnly :: S.ConfigFlags -> Bool
 execOnly flags =
-  case lookup (FlagName "execonly") (S.configConfigurationsFlags flags) of
+  case lookup (mkFlagName "execonly") (S.configConfigurationsFlags flags) of
     Just True -> True
     Just False -> False
     Nothing -> False
 
 isRelease :: S.ConfigFlags -> Bool
 isRelease flags =
-    case lookup (FlagName "release") (S.configConfigurationsFlags flags) of
+    case lookup (mkFlagName "release") (S.configConfigurationsFlags flags) of
       Just True -> True
       Just False -> False
       Nothing -> False
 
 isFreestanding :: S.ConfigFlags -> Bool
 isFreestanding flags =
-  case lookup (FlagName "freestanding") (S.configConfigurationsFlags flags) of
+  case lookup (mkFlagName "freestanding") (S.configConfigurationsFlags flags) of
     Just True -> True
     Just False -> False
     Nothing -> False
+
+#if !(MIN_VERSION_Cabal(2,0,0))
+mkFlagName :: String -> FlagName
+mkFlagName = FlagName
+#endif
 
 -- -----------------------------------------------------------------------------
 -- Clean
@@ -151,20 +156,22 @@ generateToolchainModule verbosity srcDir toolDir = do
     createDirectoryIfMissingVerbose verbosity True srcDir
     rewriteFile toolPath (commonContent ++ toolContent)
 
-idrisConfigure _ flags _ local = do
+idrisConfigure _ flags pkgdesc local = do
     configureRTS
-    generateVersionModule verbosity (autogenModulesDir local) (isRelease (configFlags local))
-    if isFreestanding $ configFlags local
-        then do
-                toolDir <- lookupEnv "IDRIS_TOOLCHAIN_DIR"
-                generateToolchainModule verbosity (autogenModulesDir local) toolDir
-                targetDir <- lookupEnv "IDRIS_LIB_DIR"
-                case targetDir of
-                     Just d -> generateTargetModule verbosity (autogenModulesDir local) d
-                     Nothing -> error $ "Trying to build freestanding without a target directory."
-                                  ++ " Set it by defining IDRIS_LIB_DIR."
-        else
-                generateToolchainModule verbosity (autogenModulesDir local) Nothing
+    withLibLBI pkgdesc local $ \_ libcfg -> do
+      let libAutogenDir = autogenComponentModulesDir local libcfg
+      generateVersionModule verbosity libAutogenDir (isRelease (configFlags local))
+      if isFreestanding $ configFlags local
+          then do
+                  toolDir <- lookupEnv "IDRIS_TOOLCHAIN_DIR"
+                  generateToolchainModule verbosity libAutogenDir toolDir
+                  targetDir <- lookupEnv "IDRIS_LIB_DIR"
+                  case targetDir of
+                       Just d -> generateTargetModule verbosity libAutogenDir d
+                       Nothing -> error $ "Trying to build freestanding without a target directory."
+                                    ++ " Set it by defining IDRIS_LIB_DIR."
+          else
+                  generateToolchainModule verbosity libAutogenDir Nothing
     where
       verbosity = S.fromFlag $ S.configVerbosity flags
       version   = pkgVersion . package $ localPkgDescr local
@@ -174,6 +181,10 @@ idrisConfigure _ flags _ local = do
       -- distribution if it's not there, so instead I just delete
       -- the file after configure.
       configureRTS = make verbosity ["-C", "rts", "clean"]
+
+#if !(MIN_VERSION_Cabal(2,0,0))
+      autogenComponentModulesDir lbi _ = autogenModulesDir lbi
+#endif
 
 idrisPreSDist args flags = do
   let dir = S.fromFlag (S.sDistDirectory flags)

--- a/Setup.hs
+++ b/Setup.hs
@@ -1,5 +1,9 @@
 {-# LANGUAGE CPP #-}
 
+#if !defined(MIN_VERSION_Cabal)
+# define MIN_VERSION_Cabal(x,y,z) 0
+#endif
+
 import Control.Monad
 import Data.IORef
 import Control.Exception (SomeException, catch)

--- a/idris.cabal
+++ b/idris.cabal
@@ -97,6 +97,14 @@ source-repository head
   type:     git
   location: git://github.com/idris-lang/Idris-dev.git
 
+custom-setup
+  setup-depends:
+    Cabal >= 1.10 && <2.1,
+    base  >= 4 && <5,
+    directory,
+    filepath,
+    process
+
 Flag FFI
   Description:  Build support for libffi
   Default:      False

